PXWeb/Code/API/Controllers/DcatController.cs

```diff
@@ -47,7 +47,8 @@ public HttpResponseMessage buildXML(string databaseType, string database, string
                 {
                     return Request.CreateResponse(HttpStatusCode.BadRequest, $"Invalid database type");
                 }
-                XML.WriteToFile(HttpContext.Current.Server.MapPath("~/dcat-ap.xml"), settings);
+                string savePath = HttpContext.Current.Server.MapPath("~/Resources/PX/Databases/" + database + "/dcat-ap.xml");
```

Comment:
Instead of hard coding it to ~/resources/PX/Databases/ use the settings object PXWeb.Settings.Current.General.Paths.PxDatabasesPath